### PR TITLE
SitesDropdown: Don't bind selectors to state

### PR DESCRIPTION
**Pull request by mcsf**
Part of #14024 

# Testing

`SitesDropdown` is used in `SiteSelectorModal` (via ellipsis actions at `/themes`), `/me/account`, `HelpContactForm`. Behaviors should remain unchanged.

### CODE REVIEW

File: client/components/sites-dropdown/index.jsx [hunk 1]
```diff
@@ -39,27 +42,23 @@ export class SitesDropdown extends PureComponent {
...
-		};
 	}
 
-	getSelectedSite() {
-		return this.props.getSite( this.state.selectedSiteSlug );
+	state = { open: false }
+
+	componentDidMount() {
+		const { initialSiteId, setSelectedSiteSlug } = this.props;
+		setSelectedSiteSlug( initialSiteId );
```
Comment (tyxla) [inline client/components/sites-dropdown/index.jsx]: It feels really odd to pass an ID to something that expects a slug ❓

Comment (mcsf) [inline client/components/sites-dropdown/index.jsx]: True; an artifact of the first pass. I should rename to `id`. :)

Comment (mcsf) [inline client/components/sites-dropdown/index.jsx]: Actually, ID on its own would clash with a prop that is part of `SitesDropdown`'s public interface, so I'm changing to `locallySelectedSiteId`.

**Comment by mcsf** [comment]
Rebased against `master`.

**Pull request merged**